emergence_lib/src/hive_mind.rs

```diff
@@ -138,20 +155,58 @@ fn place_pheromone(
     }
 }
 
-/// Toggle showing debug info   
+/// Toggle showing debug info
 fn show_debug_info(
     dev: Query<&ActionState<DevAction>, With<HiveMind>>,
     mut debug_info: ResMut<DebugInfo>,
 ) {
     let dev = dev.single();
-    let fps_info = dev.pressed(DevAction::ShowInfoText);
+    let dev_mode = dev.just_pressed(DevAction::ToggleDevMode);
+    let tile_labels = dev.just_pressed(DevAction::ToggleTileLabels);
+    let fps_info = dev.just_pressed(DevAction::ToggleInfoText);
+    let inspector = dev.just_pressed(DevAction::ToggleInspector);
+
+    // Toggle the dev mode so that what happens is intuitive to the user
+    if dev_mode {
```

Comment:
These if-blocks are a little repetitive. This might not be a problem, but I am wondering if it might make sense to extract their common logic into a function, and then call that function repeatedly?

This way, when new debug flags are added in the future, it is slightly easier to incorporate them in too.

Reply:
Agreed here.

Reply:
Ok, that makes sense. It would make it cleaner and the callback function shorter.
How does this look now?

Reply:
Yes, I think this looks better. For a moment I wondered if *this* if statement could also just become a function call, but there's enough of a difference that it doesn't make sense.